src/sqlfluff/core/dialects/base.py

```diff
@@ -230,7 +231,8 @@ def add_update_segments(self, module_dct: Dict[str, Any]) -> None:
                 if k not in self._library:
                     self.add(**{k: v})
                 else:
-                    self.replace(**{k: v})
+                    non_seg_v = cast(Union[Matchable, SegmentGenerator], v)
+                    self.replace(**{k: non_seg_v})
```

Comment:
This looks wrong. We've just asserted that it's a `BaseSegment` above, so it can't also be a `Union[Matchable, SegmentGenerator]` here. I'm assuming you put this in because you were otherwise getting errors - do you know what errors you were getting? There might be another solution.

Reply:
This must've been left over from when I was trying to get stuff to work. I removed it without issue